src/core/server/saved_objects/mappings/index_mappings.ts

```diff
@@ -19,72 +19,78 @@
 
 import { cloneDeep, isPlainObject } from 'lodash';
 
-import { formatListAsProse } from '../../utils';
-import { getRootProperties, getRootType } from './lib';
+import { EsMapping, EsMappings, getRootProperties, getRootType } from './lib';
+import { formatListAsProse } from './utils';
 
-const DEFAULT_INITIAL_DSL = {
+interface MappingExtension {
+  pluginId?: string;
+  properties: { [key: string]: EsMapping };
+}
+
+const DEFAULT_INITIAL_DSL: EsMappings = {
   rootType: {
     type: 'object',
     properties: {},
   },
 };
 
 export class IndexMappings {
-  constructor(initialDsl = DEFAULT_INITIAL_DSL, mappingExtensions = []) {
-    this._dsl = cloneDeep(initialDsl);
-    if (!isPlainObject(this._dsl)) {
+  private dsl: EsMappings = this.initialDsl;
+
+  constructor(
+    private initialDsl: EsMappings = DEFAULT_INITIAL_DSL,
+    mappingExtensions: MappingExtension[] = []
+  ) {
+    if (!isPlainObject(this.dsl)) {
```

Comment:
Still need to keep this check for people who use this library without typescript, right?

Reply:
There's a fine line between what we should and shouldn't check for. I generally lean towards relying on TypeScript, but it's fine if you want to keep this check in place.